tinygrad/codegen/lowerer.py

```diff
@@ -58,8 +70,15 @@ def lower(self, ast:UOp, opts:Renderer) -> UOp:
         self.idxs = get_grouped_dims("idx", full_shape[:global_dims], opts.global_max, reverse=True)
       else:
         # define indexes for GPU-like execution
-        self.idxs = get_grouped_dims("gidx", full_shape[:global_dims], opts.global_max, reverse=True) + \
-                    get_grouped_dims("lidx", full_shape[global_dims:first_reduce+group_for_reduces], opts.local_max)
+        reduceops = [x for x in ast.parents if x.op is UOps.REDUCE_AXIS and all(i < first_reduce+group_for_reduces for i in x.arg[1])]
+        if len(reduceops) > 1:
+          # try to reuse grouped dims across sequential reduceops
+          rdims, p = get_reduce_dims([[i-global_dims for i in r.arg[1]] for r in reduceops], full_shape[global_dims:first_reduce+group_for_reduces])
+          lidxs = get_grouped_dims("lidx", rdims, opts.local_max)
+          self.idxs = get_grouped_dims("gidx", full_shape[:global_dims], opts.global_max, reverse=True) + [lidxs[i] for i in p]
```

Comment:
here `get_grouped_dims` is called without the reused dims, then `p` assigns the `lidx`s to a place in the shape
if a dim is reused, the `lidx` will end up duplicated in `self.idxs`

that's how this PR uses the same `lidx` for sequential reduces